### PR TITLE
Only show public messages as last messages

### LINKED ISSUE #3023: Last room message should always show the last public message
Currently when a participant triggers a private command message the other participants will get an empty message in the last message of the conversation data.

## Steps to reproduce
- Create a conversation with user B
- Send _hello_
- Send _/help_
- Open Talk with user B

### Expected behaviour
The last message preview for the conversation in the list of conversations should be _hello_.

### Actual behaviour
The last message preview for the conversation in the list of conversations is empty.

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #3023

### CODE REVIEW

File: src/components/LeftSidebar/ConversationsList/Conversation.vue [hunk 1]
```diff
@@ -257,7 +257,17 @@ export default {
...
+				/**
+				 * Only use the last message as lastmessage when:
+				 * 1. It's newer than the conversations last message
+				 * 2. It's not a command reply
+				 * 3. It's not a temporary message starting with "/" which is a user posting a command
+				 */
+				if (this.messages[lastMessageId].timestamp > lastMessageTimestamp
+					&& (this.messages[lastMessageId].actorType !== 'bots'
+						|| this.messages[lastMessageId].actorId === 'changelog')
+					&& (!lastMessageId.startsWith('temp-')
```
Comment (PVince81) [inline src/components/LeftSidebar/ConversationsList/Conversation.vue:269]: you didn't check if lastMessageId is a string here but did so further below ?

you can also convert an int to a string by doing String(lastMessageId).startsWith(...)
or if too dangerous, check for string with `typeof lastMessageId === 'string'`

Comment (nickvergessen) [reply src/components/LeftSidebar/ConversationsList/Conversation.vue:269]: well it worked here, but failed on the other. Here I always got a string as id from the key list:


```
const object1 = {
  4: 'test'
};

console.log(Object.keys(object1));
// expected output: Array ["4"]
```
You can test it on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

But I can still add it if you think it's needed

Comment (PVince81) [reply src/components/LeftSidebar/ConversationsList/Conversation.vue:269]: fine to keep it as is if you're confident about the times

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by nickvergessen** [comment]
Integration test is fixed in #5542

**Pull request merged**